lib/mpool/lib/mblock_fset.c

```diff
@@ -694,6 +694,15 @@ mblock_fset_delete(struct mblock_fset *mbfsp, uint64_t *mbidv, int mbidc)
     return 0;
 }
 
+merr_t
+mblock_fset_punch(struct mblock_fset *mbfsp, uint64_t mbid, off_t off, size_t len)
+{
+    if (!mbfsp || file_id(mbid) > mbfsp->mhdr.fcnt)
```

Comment:
Similar comment here as in mblock_punch.

Reply:
Same as above, these checks can only be enforced at this layer.